text/0418-deprecate-route-render-methods.md

```diff
@@ -0,0 +1,273 @@
+        controller: 'account'
+      });
+    } else {
+      this.transitionTo('login')
+    }
+  }
+});
+```
+
+One way this could be migrated is like the following:
+
+```hbs
+{{! app/templates/authenticated.hbs}}
+
+<nav>
+  <h1>ACME Corp.</h1>
+  <section id="account-placeholder"></section>
+</nav>
+
+<section id="content">
+  {{outlet}}
+</section>
+```
+
+```hbs
+{{! app/templates/authenticated/campaigns.hbs }}
+
+{{outlet}}
+
+{{#in-element this.accountPlaceholder}}
+  {{#link-to 'account'}}
+    <img src="{{this.account.img}}" alt="{{this.account.name}} />
+  {{/link-to}}
+{{/in-element}}
+```
+
+```js
+// app/routes/authenticated.js
+import Route from '@ember/route';
+import { inject as service } from '@ember/service';
+
+export default Route.extend({
+  //...
+  user: service('user'),
+  beforeModel() {
+    if (!this.user.isLoggedIn) {
+      this.transitionTo('login')
+    }
+  }
+});
+```
+
+```js
+// app/controller/authenticated/campaigns.js
+import Route from '@ember/route';
+import Controller from '@ember/controller';
+import { inject as service } from '@ember/service';
+import { inject as controller } from '@ember/controller';
+
+export default Controller.extend({
+  //...
+  account: controller('account')
+  init() {
+    this._super(...arguments);
+    this.accountPlaceholder = document.getElementById('account-placeholder');
+  }
+});
+```
+
+If you want to do this with components you could do the same thing as the following:
+
+```hbs
+{{! app/templates/authenticated.hbs}}
+
+<nav>
+  <h1>ACME Corp.</h1>
+  <section id="account-placeholder"></section>
+</nav>
+
+<section id="content">
+  {{outlet}}
+</section>
+```
+
+```hbs
+{{! app/templates/authenticated/campaigns.hbs }}
+{{outlet}}
+
+<UserAccount />
+```
+
+```hbs
+{{! app/templates/components/user-account.hbs }}
+{{#in-element this.accountPlaceholder}}
+  {{#link-to 'account'}}
+    <img src="{{this.account.img}}" alt="{{this.account.name}} />
+  {{/link-to}}
+{{/in-element}}
+```
+
+```js
+// app/routes/authenticated.js
+import Route from '@ember/route';
+import { inject as service } from '@ember/service';
+
+export default Route.extend({
+  //...
+  user: service('user'),
+  beforeModel() {
+    if (!this.user.isLoggedIn) {
+      this.transitionTo('login')
+    }
+  }
+});
+```
+
+```js
+// app/components/user-account.js
+import Component from '@ember/route';
+import { inject as controller } from '@ember/controller';
+
+export default Component.extend({
+  // ...
+  account: controller('account'),
+  init() {
+    this._super(...arguments);
+    this.accountPlaceholder = document.getElementById('account-placeholder');
+  }
+});
+```
+
+# How We Teach This
+
+These APIs not been a mainline API for quite some time now. The guides briefly mention this functionality. In those cases we should mirgate the guides should link to the `{{in-element}}` documentation and the component documentation. The above "Transition Path" will serve as the deprecation guide.
```

Comment:
```suggestion
These APIs not been a mainline API for quite some time now. The guides briefly mention this functionality. In those cases we should migrate the guides should link to the `{{in-element}}` documentation and the component documentation. The above "Transition Path" will serve as the deprecation guide.
```

Reply:
IMO it would be helpful for this sentence to state its subject. I think @chadhietala's wording in his response to be above was just better:

> `renderTemplate` and its associated APIs do not and have not played a predominate role in Ember's common programming model for quite some time. The guides only briefly mention this functionality. In those guides cases that are documented (maybe should link to them here) the guides should link to the `{{in-element}}` documentation and component documentation as is appropriate.
>
> The "Transition Path" section above will serve as a first draft of the deprecation guide.